### PR TITLE
Fix for deck sort and berserk cards, auto select hash codes on focus

### PR DESCRIPTION
**Pull request by eMuonTau**
FIXED: User cards with Berserk skill are now using correct value to gain
attack.
FIXED: Deck builder should sort cards as seen in game, Game is sorting card ids from low to high including fused cards.
ADD: When you focus on a Deck Hash box, code will be selected automatically.

### CODE REVIEW

File: scripts/deckbuilder.js [hunk 1]
```diff
@@ -720,7 +720,7 @@ var sortDeck = function ()
         if (compare) return compare;
         compare = (cardA.type - cardB.type);
         if (compare) return compare;
-        compare = compareByID(unitA, unitB);
```
Comment (TheSench) [inline scripts/deckbuilder.js:723]: This was an intentional decision to deviate from the in-game behavior to group copies of the "same" card together.

**Review by TheSench — CHANGES_REQUESTED** [review]
I agree with the suggestions, aside from changing sorting in the DeckBuilder.  I intentionally chose to sort slightly differently from how the game does it.

**Comment by eMuonTau** [comment]
> This will sort all cards in your deck based on faction, rarity, and ID (the same way they are sorted in the game).

I got it. it is a better way of sorting. Above line in tutorial made me think it was a bug.

Sorry i forgot to create branches for each PR it would be easy merge. 

Are you planning to update user interface? I want to work on a new GUI based on material design. You are already using angularjs. I think we can create a single page app so whenever we save decks we can load decks without reloading or copying. I can create an issue ticket to discuss about user interface. Please let me know what do you think.

**Comment by TheSench** [comment]
I'm not sure how to have a discussion on here outside of these comments.  Send me a PM on Kongregate or LINE and we can chat about simulator ideas.  I fixed the berserk bug and made the suggested changes for the hash fields in the simulator/practice page.  I did not change the deckbuilder, as I often find myself wanting to actually place the cursor at specific parts of a hash, rather than copy/paste it.  I wasn't sure how to approve part of a PR, so I made the changes myself.

**Pull request closed without merging**